### PR TITLE
doc: add "Getting started" and "Rest" page

### LINKED ISSUE #102: Doc: add getting started page
This page should explains:
- How to run dotbot-controller with an nRF DK connected to the computer and a DotBot around
- Give some code snippets that allow to:
  - List available DotBots
  - Change the RGB LED on a given DotBot
  - Move a given DotBot

Maybe reuse what's available at https://github.com/DotBots/PyDotBot/wiki/HandsOn-Tutorial

### PR DESCRIPTION
**Pull request by aabadie**
fixes #102

### CODE REVIEW

File: doc/getting_started.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Getting started
+
+This document will guide through setting up PyDotBot connected to a DotBot
```
Comment (filmak) [inline doc/getting_started.md:3]: This document will help guide you through the setup of PyDotBot connected to a DotBot

File: doc/getting_started.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+gateway and a DotBot robot.
+
+## Prerequisites
+
+Make sure you have access to an nRF DK board (nrf52833dk, nrf52840dk or
+nrf5340dk) and to a DotBot (v1 or v2).
+
+Follow the instructions of the
```
Comment (filmak) [inline doc/getting_started.md:11]: Follow the instructions in the

File: doc/getting_started.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+## Setup the gateway
+
+The gateway is used to bridge the UART communication between PyDotBot running on
+a computer and the BLE radio used to communicate wirelessly with the DotBot(s).
+
+1. Connect the nRF DK gateway to your computer
+
+2. Identify the TTY port it is connected to. On Linux, it should be `/dev/ttyACM0`.
+  On Windows, check the device manager, it should be `COM1`, `COM2`, `COM3`, etc
```
Comment (filmak) [inline doc/getting_started.md:30]: maybe a technical note, but sometimes DKs show up as two difference COM ports?

File: doc/getting_started.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+3. From a terminal, run `dotbot-controller` with the TTY port you identified
```
Comment (filmak) [inline doc/getting_started.md:32]: "From a terminal" doesn't work for windows, I think. Does powershell work?

File: doc/getting_started.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  above:
+
+```
+dotbot-contoller --port <tty port>
+```
+
+At this point, if the DotBot is powered on with fully charged batteries, you
+should see it poping up in the logs, with a similar output:
```
Comment (filmak) [inline doc/getting_started.md:40]: you should see an output in the logs that looks something like:

File: doc/getting_started.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+2023-11-29T07:55:11.725907Z [info     ] Lighthouse initialized         [pydotbot] context=dotbot.lighthouse2
+2023-11-29T07:55:11.726325Z [info     ] Controller initialized         [pydotbot] context=dotbot.keyboard
+2023-11-29T07:55:11.726746Z [info     ] Starting web server            [pydotbot] context=dotbot.server
+2023-11-29T07:55:11.739085Z [info     ] Serial port thread started     [pydotbot] context=dotbot.serial_interface
+2023-11-29T07:55:12.197714Z [info     ] New dotbot                     [pydotbot] application=DotBot context=dotbot.controller msg_id=90350129 payload_type=ADVERTISEMENT source=9903ef26257feb31
+```
+
+## Control your DotBot
+
+1. Open [http://localhost:8000/dotbots](http://localhost:8000/dotbots) and you
```
Comment (filmak) [inline doc/getting_started.md:53]: In a browser, open ...

File: doc/getting_started.md (hunk 3 above)
Comment (aabadie) [reply doc/getting_started.md:30]: yes the nrf53dk shows 2 COM ports, I don't which one is the right one on windows. On Linux it's the lowest value among `/dev/ttyACMx` and `/dev/ttyACMy`

File: doc/rest.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
+# REST
+
+When running and connected to a DotBot gateway, the `dotbot-controller`
```
Comment (filmak) [inline doc/rest.md:3]: While connected to a DotBot gateway, the `dotbot-controller`

File: doc/rest.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+application provides a REST server to get information and control connected
```
Comment (filmak) [inline doc/rest.md:4]: application provides a REST server to send commands to and receive information from connected DotBots.

File: doc/rest.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+DotBots.
+
+The REST API is documented in the running `dotbot-controller` application itself
+at [http://localhost:8000/api](http://localhost:8000/api). This page also allows
```
Comment (filmak) [inline doc/rest.md:8]: This page also allows you

File: doc/rest.md [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+:class: bg-primary
+:width: 700px
+:align: center
+```
+
+## Prerequisites
+
+Make sure you already followed the [getting started](getting_started) page and
+have a functional setup with `dotbot-controller` running and connected to a
+nRF DK gateway.
```
Comment (filmak) [inline doc/rest.md:22]: This is fine, but I would reverse the two phrases. It's stylistic, though, not necessary

File: doc/rest.md [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+    "last_seen": 1701244665.8099585,
+    "waypoints": [],
+    "waypoints_threshold": 40,
+    "position_history": []
+  }
+]
+```
+
+This is a list of all DotBots connected to the `dotbot-controller`. In the
+example above, there's only one DotBot connected.
```
Comment (filmak) [inline doc/rest.md:70]: there's -> there is

File: doc/rest.md [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+The `address` is an 8 bytes data that uniquely identifies a DotBot in the
```
Comment (filmak) [inline doc/rest.md:71]: The 8-byte `address` uniquely identifies a DotBot in the

File: doc/rest.md [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+controller. The `status` tells if the DotBot is `Alive` (0), `Lost` (1) or
```
Comment (filmak) [inline doc/rest.md:72]: The `status` indicates whether

File: doc/rest.md [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+`Dead` (2).
```
Comment (filmak) [inline doc/rest.md:73]: RIP

File: doc/rest.md [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+
+If the DotBot `address` is already known, use the `contoller/dotbots/<address>`
```
Comment (filmak) [inline doc/rest.md:75]: A bit ambiguous what a "known" address means. Known to whom?

File: doc/rest.md [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+{
+  "left_x": 0,
+  "left_y": 0,
+  "right_x": 0,
+  "right_y": 0
+}
+```
+
+For the DotBot application type, only left_y and right_y values are useful,
+left_x and right_x being ignored by the firmware running on the DotBots.
```
Comment (filmak) [inline doc/rest.md:134]: I don't understand this sentence.

File: doc/rest.md (hunk 14 above)
Comment (aabadie) [reply doc/rest.md:73]: :skull:

File: doc/rest.md (hunk 15 above)
Comment (aabadie) [reply doc/rest.md:75]: known to the controller

File: doc/rest.md (hunk 16 above)
Comment (aabadie) [reply doc/rest.md:134]: I reworded it

File: doc/rest.md (hunk 15 above)
Comment (aabadie) [reply doc/rest.md]: I detailed a bit more

File: doc/rest.md (hunk 10 above)
Comment (aabadie) [reply doc/rest.md:22]: Let's keep it like this

File: doc/getting_started.md [hunk 17]
```diff
@@ -0,0 +1,85 @@
+# Getting started
+
+This document will help guide you through the setup of PyDotBot connected to
+a DotBot gateway and a DotBot robot.
+
+## Prerequisites
+
+Make sure you have access to an nRF DK board (nrf52833dk, nrf52840dk or
+nrf5340dk) and to a DotBot (v1 or v2).
```
Comment (filmak) [inline doc/getting_started.md:9]: capitalized RF and DK, I think?

File: doc/getting_started.md [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+
+Follow the instructions in the
+[DotBot firmware getting started page][dotbot-firmware-getting-started].
+
+## Install PyDotBot
+
+Use pip to install the latest version of PyDotBot from [pypi][pydotbot-pypi]:
```
Comment (filmak) [inline doc/getting_started.md:16]: maybe you need to specify valid python versions? I don't know.

File: doc/getting_started.md [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+
+```{image} _static/images/pydotbot-ui-active.png
+:alt: Single active DotBot item, with widgets
+:class: bg-primary
+:width: 400px
+:align: center
+```
+
+4. Check that you can control the DotBot:
+  - Move it using the virtual joystick in the UI
```
Comment (filmak) [inline doc/getting_started.md:78]: by clicking on the joystick and dragging it in the direction that you want the DotBot to move

File: doc/getting_started.md [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+  - Move it using the arrow keys of your keyboard
```
Comment (filmak) [inline doc/getting_started.md:79]: Control it...

File: doc/getting_started.md [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+  - Change RGB LED color by pressing "b", "g" or "b" or by using the color picker
```
Comment (filmak) [inline doc/getting_started.md:80]: you did not fix this one.

File: doc/rest.md [hunk 22]
```diff
@@ -0,0 +1,173 @@
...
+[requests](https://pypi.org/project/requests/) package. You can install it on
+your computer using pip:
+
+```
+pip install -U requests
+```
+
+## The basics
+
+First let's start by fetching the information about available DotBots using
```
Comment (filmak) [inline doc/rest.md:34]: First, let's ...

File: doc/rest.md [hunk 23]
```diff
@@ -0,0 +1,173 @@
...
+    "waypoints_threshold": 40,
+    "position_history": []
+  }
+]
+```
+
+This is a list of all DotBots connected to the `dotbot-controller`. In the
+example above, there is only one DotBot connected.
+The 8-byte `address` uniquely identifies a DotBot in the controller. The
+`status` indicates whether the DotBot is `Alive` (0), `Lost` (1) or `Dead` (2).
```
Comment (filmak) [inline doc/rest.md:72]: maybe you should write what alive lost and dead actually mean

File: doc/rest.md [hunk 24]
```diff
@@ -0,0 +1,173 @@
...
+
+If the DotBot `address` is already known by the controller, e.g. it identifies
+one of the DotBots returned by the previous request, use the
```
Comment (filmak) [inline doc/rest.md:75]: returned by a previous request

File: doc/getting_started.md (hunk 18 above)
Comment (aabadie) [reply doc/getting_started.md:16]: The supported Python versions are documented in the package itself normally

**Comment by aabadie** [comment]
Thank you for the reviews @filmak !

**Pull request merged**